### PR TITLE
loadtest: add support for push metrics

### PR DESCRIPTION
**Pull request by calvinrzachman**
Equip the test orchestrator with the ability to push metrics on execution time to a configurable remote prometheus push gateway.

Related to: https://github.com/lightninglabs/taproot-assets/pull/662 (`loadtest` metrics vs. `tapd` metrics)

### CODE REVIEW

File: itest/loadtest/config.go [hunk 1]
```diff
@@ -44,6 +45,15 @@ type BitcoinConfig struct {
 	TLSPath  string `long:"tlspath" description:"Path to btcd's TLS certificate, if TLS is enabled"`
 }
 
+// PrometheusGatewayConfig defines exported config options for connecting to the
+// Prometheus PushGateway.
+type PrometheusGatewayConfig struct {
+	Enabled bool   `long:"enabled" description:"Enable pushing metrics to Prometheus PushGateway"`
+	Host    string `long:"host" description:"Prometheus PushGateway host address"`
+	Port    int    `long:"port" description:"Prometheus PushGateway port"`
+	PushURL string
```
Comment (ffranr) [inline itest/loadtest/config.go:54]: nit: I think you should add a doc comment to describe `PushURL`.

**Review by ffranr — COMMENTED** [review]
Nice! Good to see an example of how this is done.

We usually hard wrap at 80 characters, I think some lines go over right now. 

I think you should squash these commits.

**Comment by dstadulis** [comment]
Other metric criteria to collect:
- Universe
- Balance
- Number of UTXOs
- Queries per second QPS
- Aspects exposed on the RPC
- Time to fetch proof
- "Delay metrics"
- Other criteria to make alerts

End goal:
- Loop-service-like metrics
- Sanity checks

Comments:
- Avoid high cardinality metrics
- Collector is pased an interface

**Review by GeorgeTsagk — COMMENTED** [review]
Closing this in favor of #737

**Pull request closed without merging**